config.go

```diff
@@ -21,6 +21,7 @@ type configOption struct {
 	listener        interface{}
 	storage         Storage
 	httpClient      *http.Client
+	customHeaders	http.Header
```

Comment:
A very reasonable expectation, but `http.Header` is a [builtin](https://golang.org/pkg/net/http/#Header) type which is defined as `map[string][]string`. That's a bit ugly but it means that the `Header` type is a map of string arrays. In Javascript, it would look like this:

```js
const header = {
  'Foo': ['bar', 'baz']
}
```
I don't really know why they chose to use an array for the values instead of just a string, but the HTTP spec says multiple headers can have the same name if the values can be combined so maybe that is why. Rarely used though.